### PR TITLE
Eli feat deploy

### PR DESCRIPTION
**Pull request by elfisher**
# Deploy Script

This PR adds a deploy script to MiraKit that enables developers to easily deploy their applications to the Mira Platform.

### CODE REVIEW

File: scripts/deploy.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    fs.readFile(fileUri, (err, data) => {
+      if(err){
+        console.log(err);
+        reject(err);
+      }
+      request({
+        method: "PUT",
+        url: presignedUrl,
+        body: data
+      }, function(err, res, body){
```
Comment (KiaFathi) [inline scripts/deploy.js]: Mixing callbacks and promises is a bad practice.

Comment (KiaFathi) [inline scripts/deploy.js]: Mixing callbacks and promises is a bad practice.

Comment (KiaFathi) [inline scripts/deploy.js]: https://spin.atomicobject.com/2017/04/06/nodejs-promises-callbacks/

File: scripts/deploy.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+})
+.then(() => {
+  if (thumbnailExists) {
+    return uploadPresentationFile(thumbnailPath, thumbnailUrl);
+  }
+  console.warn('WARNING: No thumbnail was uploaded');
+  return;
+}).then(() => uploadPresentationFile(bundlePath, sourceUrl)
+).catch(err => {
+	console.log(err);
```
Comment (KiaFathi) [inline scripts/deploy.js]: We should throw an error here.

File: scripts/deploy.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  iconUrl = response.icon_url;
+  sourceUrl = response.source_url;
+  thumbnailUrl = response.thumbnail_url;
+})
+.then(() => {
+  if (iconExists) {
+    return uploadPresentationFile(iconPath, iconUrl);
+  }
+  console.warn('WARNING: No icon was uploaded');
```
Comment (KiaFathi) [inline scripts/deploy.js]: Should it throw an error here?

File: scripts/deploy.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  return;
+})
+.then(() => {
+  if (thumbnailExists) {
+    return uploadPresentationFile(thumbnailPath, thumbnailUrl);
+  }
+  console.warn('WARNING: No thumbnail was uploaded');
```
Comment (KiaFathi) [inline scripts/deploy.js]: Should it throw an error here?

File: scripts/deploy.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    body: JSON.stringify({
+      version,
+      author_id: authorId,
+      name,
+      presentation_properties,
+      strings})
+  })
+}).then(response => {
+    return response.json();
+}
```
Comment (KiaFathi) [inline scripts/deploy.js]: Please format code.

File: scripts/deploy.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
+#! /usr/bin/env node
```
Comment (KiaFathi) [inline scripts/deploy.js]: Run this file through prettier.

File: README.md [hunk 7]
```diff
@@ -205,49 +206,72 @@ To set this up you will need to:
...
-// END: Done by User
-simulator(App, definition);
-```
+   ``` js
+   import React from 'react';
+   import ReactDOM from 'react-dom';
+   import App from '../src';
+   import definition from '../definition.json';
+   import { simulator } from 'mira-kit';
+   // END: Done by User
```
Comment (KiaFathi) [inline README.md]: You can remove this comment here.

File: scripts/deploy.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+// NOTE: thumbnail.svg is optional
+if (!fs.existsSync(thumbnailPath)) {
+  thumbnailExists = false;
+  console.warn('WARNING: no thumbnail.svg found for project');
+}
+
+// Get app version and definition properties
+const  { version } = packageJson;
```
Comment (KiaFathi) [inline scripts/deploy.js]: spacing here?

File: scripts/deploy.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  })
+}).then(response => {
+    return response.json();
+}
+).then(response => {
+  const { deployment_id: deploymentId } = response;
+  if (!deploymentId) {
+    throw new Error(JSON.stringify(response));
+  }
+  iconUrl = response.icon_url;
```
Comment (KiaFathi) [inline scripts/deploy.js]: Globals from a promise are not a good practice.

File: scripts/deploy.js [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+const thumbnailPath = path.join(distDir, 'thumbnail.svg');
+
+// Pull definition and package JSON
+const definitionPath = path.join(processDir, 'definition.json');
+const packagePath = path.join(processDir, 'package.json');
+const definition = require(definitionPath);
+const packageJson = require(packagePath);
+
+// assume optional files exist, but if they don't save they don't.
+let thumbnailExists = true;
```
Comment (KiaFathi) [inline scripts/deploy.js]: Should assign these values to the synchronous calls you are doing anyways.

File: scripts/deploy.js [hunk 11]
```diff
@@ -0,0 +1,145 @@
+#! /usr/bin/env node
+
+const path = require('path');
+const fs = require('fs');
+const fetch = require('isomorphic-fetch');
```
Comment (KiaFathi) [inline scripts/deploy.js]: Let's not use both fetch and request here.

File: scripts/deploy.js (hunk 4 above)
Comment (elfisher) [inline scripts/deploy.js]: for now these are optional. Should we make them required?

File: scripts/deploy.js (hunk 3 above)
Comment (elfisher) [inline scripts/deploy.js]: only bundle is required for now. Should we make everything required?

**Pull request merged**